flink-optimizer/src/main/java/org/apache/flink/optimizer/operators/ReduceProperties.java

```diff
@@ -66,30 +70,62 @@ public SingleInputPlanNode instantiate(Channel in, SingleInputNode node) {
 		}
 		else {
 			// non forward case. all local properties are killed anyways, so we can safely plug in a combiner
```

Comment:
The `else` branch will not be entered if the Reduce's predecessor is a Partition operator.
You need to add an `if else` branch to the condition.